### ISSUE TITLE
mdl_events_queue being filled up and failing to clear out.

### ISSUE THREAD

**Issue by Syxton** [turnitin/moodle-plagiarism_turnitin#50]
We are having issues with the events_queue table filling up and becoming clogged by events in the queue that don't seem to have submissions.  The event data points to assignments that don't have submissions and the userid given is for the instructor.  Very strange, but it is causing a backup in our events queue.

**Comment by jmcgettrick** [comment]
Hi @Syxton, we've seen a few people having issues like this. Usually, the first submission in the queue is the problem because the assignment itself can't be created or changed in Moodle. Do you have any Acitvity logs or API logs to consult as these should point out what the error is?

Moodle does create an event in the events_queue when a module is created or updated so maybe this is one that's causing an issue.

We're working on getting this problem fixed so that all submission events get processed and the ones that don't get sent because of assignment problems will be listed in the errors page that can be resubmitted.

**Comment by Syxton** [comment]
the first submission in the queue is always the issue...because it doesn't get cleared and causes all the rest to never get processed.  Currently we are having to clear out events daily.  I am currently testing some ideas, hopefully I'll know more soon.

**Comment by Syxton** [comment]
We are currently stepping though the events and it is stopping frequently.  I have a print out of our turnitin events.  I'm marking the rows that it stops on.  I'll submit the file when I think there is enough data.

**Comment by Syxton** [comment]
The cause seems to be that new assignments aren't getting a turnitin assignment created.
This module could not be created on
Turnitin, please consult your API logs for further information

Our api logs are empty, even with 
Enable Diagnostic Mode is enabled 
Enable Network Performance Logging is enabled

**Comment by Syxton** [comment]
the logs work on our dev system

2015-12-07 11:52:52 -0500 (REQUEST) - Turnitin Assignment part updated - id: 43183995
2015-12-07 11:53:04 -0500 (API_ERROR) - There was an error trying to create the assignment in Turnitin<br/><br/>Fault Code: invaliddata | File: /var/www/moodle/mod/turnitintooltwo/sdk/assignmentsoap.class.php | Line: 33 | Message: submit_papers_to - invalid value 
 | Code: 0

**Comment by Syxton** [comment]
------------------------------------------------------------
2015-12-07 11:53:04 -0500 - INFO -->
------------------------------------------------------------
Content-type: text/xml;charset="utf-8"
Accept: text/xml
Cache-Control: no-cache
Pragma: no-cache
SOAPAction: "http://www.imsglobal.org/soap/lis/oms1p0/createByProxyLineItem"
Content-length: 5391
X-Integration-Version: 2015051103
X-Plugin-Version: 2015040111
Source: 12
Authorization: OAuth oauth_nonce="43c134707a1c61e6f780794e7d631b4165cc19b3", oauth_timestamp="1449507184", oauth_consumer_key="81460", oauth_signature_method="HMAC-SHA1", oauth_version="1.0", oauth_body_hash="z6jYOaiQEYJxUpX%2FP4YQbHBQt7o%3D", oauth_signature="nbtN6In2lPhqs5ueq%2FfXwYwA4xY%3D"
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="http://www.imsglobal.org/services/lis/oms1p0/wsdl11/sync/imsoms_v1p0"><SOAP-ENV:Header><ns1:imsx_syncRequestHeaderInfo>
    <ns1:imsx_version>V1.0</ns1:imsx_version>
    <ns1:imsx_messageIdentifier>a8aaaecb-f450-4155-9281-7c385d607674</ns1:imsx_messageIdentifier>
</ns1:imsx_syncRequestHeaderInfo>
</SOAP-ENV:Header><SOAP-ENV:Body><ns1:createByProxyLineItemRequest><ns1:lineItemRecord><ns1:sourcedGUID><ns1:sourcedId/></ns1:sourcedGUID><ns1:lineItem><ns1:context><ns1:contextIdentifier>9750818</ns1:contextIdentifier><ns1:contextType>courseSection</ns1:contextType></ns1:context><ns1:label>Another assignment with Turnitin (Moodle PP)</ns1:label><ns1:extension><ns1:extensionNameVocabulary>http://www.turnitin.com/static/source/media/turnitinvocabularyv1p0.xml</ns1:extensionNameVocabulary><ns1:extensionValueVocabulary>http://www.imsglobal.org/vdex/lis/omsv1p0/extensionvocabularyv1p0.xml</ns1:extensionValueVocabulary><ns1:extensionField><ns1:fieldName>StartDate</ns1:fieldName><ns1:fieldType>DateTime</ns1:fieldType><ns1:fieldValue>2015-12-07T05:00:00Z</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>DueDate</ns1:fieldName><ns1:fieldType>DateTime</ns1:fieldType><ns1:fieldValue>2015-12-14T05:00:00Z</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>FeedbackReleaseDate</ns1:fieldName><ns1:fieldType>DateTime</ns1:fieldType><ns1:fieldValue>2015-12-07T05:00:00Z</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>AuthorOriginalityAccess</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>RubricId</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue></ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>SubmittedDocumentsCheck</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>1</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>InternetCheck</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>1</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>PublicationsCheck</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>1</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>MaxGrade</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>100</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>LateSubmissionsAllowed</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>1</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>SubmitPapersTo</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>ResubmissionRule</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>1</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>BibliographyExcluded</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>QuotedExcluded</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>SmallMatchExclusionType</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>SmallMatchExclusionThreshold</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>Erater</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterSpelling</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterGrammar</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterUsage</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterMechanics</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterStyle</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterSpellingDictionary</ns1:fieldName><ns1:fieldType>String</ns1:fieldType><ns1:fieldValue>en_US</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>EraterHandbook</ns1:fieldName><ns1:fieldType>Integer</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>TranslatedMatching</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField><ns1:extensionField><ns1:fieldName>AllowNonOrSubmissions</ns1:fieldName><ns1:fieldType>Boolean</ns1:fieldType><ns1:fieldValue>0</ns1:fieldValue></ns1:extensionField></ns1:extension></ns1:lineItem></ns1:lineItemRecord></ns1:createByProxyLineItemRequest></SOAP-ENV:Body></SOAP-ENV:Envelope>

------------------------------------------------------------



------------------------------------------------------------
2015-12-07 11:53:04 -0500 - INFO -->
------------------------------------------------------------
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
  <SOAP-ENV:Header>
    <tns:imsx_syncResponseHeaderInfo xmlns:tns="http://www.imsglobal.org/services/lis/oms1p0/wsdl11/sync/imsoms_v1p0">
      <tns:imsx_version>V1.0</tns:imsx_version>
      <tns:imsx_messageIdentifier>FC03A72C-9D02-11E5-B0BE-33137A307A57</tns:imsx_messageIdentifier>
      <tns:imsx_statusInfo>
        <tns:imsx_codeMajor>failure</tns:imsx_codeMajor>
        <tns:imsx_severity>error</tns:imsx_severity>
        <tns:imsx_messageRefIdentifier>a8aaaecb-f450-4155-9281-7c385d607674</tns:imsx_messageRefIdentifier>
        <tns:imsx_description>submit_papers_to - invalid value 
</tns:imsx_description>
        <tns:imsx_codeMinor>
          <tns:imsx_codeMinorField>
            <tns:imsx_codeMinorFieldName>TargetEndSystem</tns:imsx_codeMinorFieldName>
            <tns:imsx_codeMinorFieldValue>invaliddata</tns:imsx_codeMinorFieldValue>
          </tns:imsx_codeMinorField>
        </tns:imsx_codeMinor>
      </tns:imsx_statusInfo>
    </tns:imsx_syncResponseHeaderInfo>
  </SOAP-ENV:Header>
  <SOAP-ENV:Body>
    <tns:createByProxyLineItemResponse xmlns:tns="http://www.imsglobal.org/services/lis/oms1p0/wsdl11/sync/imsoms_v1p0"/>
  </SOAP-ENV:Body>
</SOAP-ENV:Envelope>

**Comment by Syxton** [comment]
[apilog.txt](https://github.com/turnitin/moodle-plagiarism_turnitin/files/54297/apilog.txt)

**Comment by Syxton** [comment]
So it appears there is a problem with PP not allowing "No Repository" to be selected.

**Comment by jmcgettrick** [comment]
Hi @Syxton, it appears your admin has your Turnitin account set to "submit all papers to the standard repository." This can be edited via the website or you can change the Moodle setting "Paper Repository Assignments" in Site Admin > Turnitin Assignment 2 to match it and that should remove that issue.

**Comment by Syxton** [comment]
We were using the setting "Enable instructor standard repository options".  Once we discovered that it was the submit_papers_to field causing issues, we tried switching to "Submit all papers to standard repository" but that failed as well.  So we switched back to "Enable instructor standard repository" and set "Standard Repository" as the default and locked it.  This seems to work.  However, I find it hard to believe that the failed events are the proper response to a 0 value.
